index.js

```diff
@@ -17,29 +17,30 @@ module.exports = function kindOf(val) {
   if (val === true || val === false || val instanceof Boolean) {
     return 'boolean';
   }
+
   if (typeof val !== 'object') {
+    // here: function, symbol, string (literal), number (literal)
     return typeof val;
   }
   if (Array.isArray(val)) {
     return 'array';
   }
-
-  var type = toString.call(val);
-
-  if (val instanceof RegExp || type === '[object RegExp]') {
+  if (val instanceof RegExp) {
     return 'regexp';
   }
-  if (val instanceof Date || type === '[object Date]') {
+  if (val instanceof Date) {
```

Comment:
this will need to be changed back. same with regex. `instanceof` fails when the object originates from a different javascript context. here is a good writeup about some of the failings of `instanceof` http://bonsaiden.github.io/JavaScript-Garden/#types.typeof

the rest of the pr is great, thanks!